modules/ROOT/pages/architecture/hexagonale_architecture.adoc

```diff
@@ -194,6 +194,180 @@ This makes the application more difficult to understand and harder to locate the
+.Module-based hexagonal component architecture with multiple modules
+image::hexagonal_component_architecture_modules2.drawio.svg["devonfw hexagonal architecture modules",scaledwidth="80%",align="center"]
+
+A crucial aspect of several business components is their communication with each other.
+For example, it is necessary for business component C1 to communicate with component C2.
+For this you have to follow these steps:
+
+* a Maven dependency from adapter C1 to adapter C2 need to be added
+* an `JVM In adapter` is created in component C2, which provides an interface, defines transport objects (TO) and enables these TOs to be mapped to the domain objects in the core of component C2.
+* an `JVM out adapter` is created in component C1, which maps the domain objects from the core of component C1 into a transport object (defined in adapter C2, see previous step) and then calls the interface of adapter C2.
+
+[source,plaintext]
+----
+application A1
+├── bootstrap (module)
+├── component C1 (module)
+│   ├── core (module)
+│   │   └── ...
+│   └── adapter (module)
+│       ├── in
+│       │   └── web
+│       │       └── ...
+│       └── out
+│           ├── repository
+│           │   └── ...
+│           └── jvm
+│               └── component C2
+│                   ├── mapper
+│                   │   └── BookingMapper.java
+│                   └── BookingComponentC1OutAdapter.java
+└── component C2 (module)
+    ├── core (module)
+    │   └── ...
+    └── adapter (module)
+        ├── in
+        │   ├── event
+        │   │   └── ...
+        │   └── jvm
+        │       └── component C1
+        │           ├── model
+        │           │   └── BookingUpdateDto.java
+        │           └── BookingComponentC1InAdapter.java
+        └── out
+            └── repository
+                └── ...
+----
+
+In this way, the two business components can communicate with each other within a modular architecture.
+At first glance, this seems complex and may involve overhead, since, for example, two mappings have to be implemented.
+However, this is deliberately chosen because, on the one hand, it means that communication between business components is used consciously and only where really necessary.
+On the other hand, these adapters represent a predetermined breaking point for later division.
+More on this in the next section.
+
+=== Extract business components into microservices
+
+Increasing technical complexity or changing scaling requirements may make it necessary to (partially) convert the existing module structure into a microservice architecture over time.
+The previously selected structure of the hexagonal architecture and the explicit communication via JVM adapters enable such a transformation with relatively little effort.
+The decisive advantage is that communication takes place beforehand via firmly defined interfaces.
+These now have to be switched from a communication technology based purely on JVM means to a communication technology.
```

Comment:
I think it makes sense to explain this pure JVM techniques with an example.
E.g. direct procedure call from the outgoung adapter C1 to the incoming adpter of C2
or a internal messaging provider like... Quarkus or Sprin possibilities here